### PR TITLE
Process Data in Models

### PR DESCRIPTION
**Pull request by studiosnabeel**
### Create custom methods
**Users**

- A method that returns the 3 most recent posts for a given user.

**Posts**

- A method that updates the posts counter for a user.
- A method that returns the 5 most recent comments for a given post.

**Comments**

- A method that updates the comments counter for a post.

**Likes**

- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
   has_many :comments
   has_many :likes
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+
+  def count_updater(author_id)
+    user = User.find_by(id: author_id)
+    user.posts_counter = Post.where(author_id:).count
+    user.save
+  end
```
Comment (bobb-Rob) [inline app/models/post.rb]: Well done on implementing the methods required for this project. 

- [ ] I notice that you have not called this **_count_updater_** method. You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#count_updater` method every time a new post has been saved to the database, then make the count_updater` method private. 
- [ ] Also, the code in this method _**count_updater**_ will have performance issues but you can use a more direct and straightforward code as shown below  

See example 👇🏾 
```ruby
# app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,2 +1,10 @@
 class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def count_updater(post_id)
+    post = Post.find_by(id: post_id)
+    post.comments_counter = Comments.where(Post_id:).count
+    post.save
+  end
```
Comment (bobb-Rob) [inline app/models/comment.rb]: - [ ] This method can be refactored to improve performance. See my suggestion here 👇🏾 
 You can also call this method using the `after_save` method as I mentioned in my previous comment.

```ruby
# app/models/comment.rb
...

def count_updater # the most suitable name for this method should be -> update_comments_counter
  post.increment!(:comments_counter)
end
```

File: app/models/like.rb [hunk 3]
```diff
@@ -1,2 +1,9 @@
 class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def count_updater(post_id)
+    post = Post.find_by(id: post_id)
+    post.likes_counter = likes.where(post_id:).count
+  end
```
Comment (bobb-Rob) [inline app/models/like.rb]: The same thing occurs here. Please remember to use the `after_save` callback to call this method once a `like` is created and saved.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @studiosnabeel, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @studiosnabeel 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**